src/app/components/file-uploader/drag-and-drop.directive.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  Directive,
+  Output,
+  Input,
+  EventEmitter,
+  HostBinding,
+  HostListener
+} from '@angular/core';
+
+@Directive({
+  selector: '[app-drag-and-drop]'
+})
+export class DndDirective {
```

Comment:
Please rename this to DragAndDropDirective. The name 'DndDirective' might be easy to remember now, but in the long run when the project gets bigger and bigger, this name will become more and more confusing.